packages/rosmsg-deser/src/DeserializerFactory.ts

```diff
@@ -0,0 +1,319 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+import { RosMsgDefinition, RosMsgField } from "rosbag";
+
+import deserializers from "./BuiltinDeserialize";
+
+// Sizes for builtin types - if a type has a fixed size, the deserializer is able to optimize
+const fixedSizeTypes = new Map<string, number>([
```

Comment:
this is probably not a big deal, but if I had to guess I would think plain objects would be faster than Map for small amounts of data, since they have been heavily optimized for years?

Reply:
This will enforce that for each key `F` used in the Map, `deserializers.${F}Array` must exist. Should catch the missing `boolArray` and `durationArray` cases (assuming the Map constructor is typed correctly - if not, the same solution can be used with an object type)

```suggestion
type ArrayTypes<Keys> = Keys extends `${infer T}Array` ? T : never;
const fixedSizeTypes = new Map<ArrayTypes<keyof typeof deserializers>, number>([
```

Reply:
Breaks `fixedSizeTypes.get(field.type)` elsewhere in the code since field.type is a string (RosMsgField interface defines that).